### DISCUSSION TITLE
Ripgrep with multiple pattern

### DISCUSSION THREAD

**Discussion by radodrabik** [BurntSushi/ripgrep#1976] [Q&A] [answered]
Hello, i am trying to use ripgrep with multiple patterns. I can achieve that with pattern1|pattern2 or the -e option.

My problem is that i use the --json  output and i want to know which pattern (pattern 1 or pattern2) matched my line. 

Is this somehow achievable? Thank you for you input.

**Comment by BurntSushi** [comment, marked as answer]
No, unfortunately ripgrep does not expose which pattern matched. It's possible this could be added some day, but it is quite a bit of work. There is also API design, since users could reasonably expect different semantics depending on what they're doing. For example, you might want to report all possible matches, even if they're overlapping. Or just report matches in the order in which patterns are given. If a pattern matches, then no other pattern could match at that span.
